include/onnxruntime/core/platform/ort_mutex.h

```diff
@@ -186,4 +186,11 @@ std::cv_status OrtCondVar::wait_for(std::unique_lock<OrtMutex>& cond_mutex,
   return steady_clock::now() - steady_now < rel_time ? std::cv_status::no_timeout : std::cv_status::timeout;
 }
 };  // namespace onnxruntime
+#else
+#include <mutex>
+#include <condition_variable>
+namespace onnxruntime {
+using OrtMutex = std::mutex;
```

Comment:
It's dangerous and may not work. Some of our code was written in a subtle way that it assumes OrtMutex can be constructed at compile time. But std:mutex doesn't.  If you replace OrtMutex with std::mutex, you could hit initialize order problem. But they don't always happen and very hard to debug or find.

Reply:
this change is not expected to happen when not building wasm. reverted this back to checking macro `__wasm__`.

also, currently multithreading is not supported yet in wasm. implementation of OrtMutex for WASM depends on further investigations on emscripten pthread support.